biz.aQute.repository/src/aQute/bnd/repository/maven/provider/IndexFile.java

```diff
@@ -13,403 +11,527 @@
+				parseSingle(archive, file);
 			}
 		} catch (Exception e) {
-			e.printStackTrace();
-			descriptor.error = e.toString();
-			refresh.set(true);
 
+			// TODO
+			e.printStackTrace();
 		}
 	}
 
-	private void saveDescriptor(BundleDescriptor descriptor) throws IOException, Exception {
-		File df = getDescriptorFile(descriptor.archive);
-		IO.mkdirs(df.getParentFile());
-		CODEC.enc()
-			.to(df)
-			.put(descriptor);
+	private boolean isMulti(String name) {
+		if (multi == null)
+			return false;
+
+		String[] extension = Strings.extension(name.toLowerCase());
+		return extension.length == 2 && Strings.in(multi, extension[1]);
 	}
 
-	private void loadDescriptorAsync(final Archive archive) throws Exception {
-		if (updateLocal(archive))
-			return;
+	private void parseSingle(Archive archive, File single) throws Exception {
+		ResourceBuilder rb = new ResourceBuilder();
+		rb.addFile(single, single.toURI());
+
+		Optional<Capability> identity = rb.getCapabilities()
+			.stream()
+			.filter(c -> c.getNamespace()
+				.equals(IdentityNamespace.IDENTITY_NAMESPACE))
+			.findFirst();
 
-		updateAsync(archive);
+		if (!identity.isPresent()) {
+			String name = archive.getWithoutVersion();
+			MavenVersion version = archive.revision.version;
+
+			BridgeRepository.addInformationCapability(rb, name, version.getOSGiVersion(), repo.toString(),
+				"Not a bundle");
+		}
+
+		Resource resource = rb.build();
+		synchronized (this) {
+			put(archive, resource);
+		}
 	}
 
-	private boolean updateLocal(final Archive archive) {
-		File archiveFile = repo.toLocalFile(archive);
-		if (archiveFile.isFile()) {
-			File descriptorFile = getDescriptorFile(archive);
-			if (descriptorFile.isFile()) {
-				try {
-					BundleDescriptor descriptor = CODEC.dec()
-						.from(descriptorFile)
-						.get(BundleDescriptor.class);
-					descriptor.promise = promiseFactory.resolved(archiveFile);
-					descriptor.resource = null;
-					descriptors.put(archive, descriptor);
-					return true;
-				} catch (Exception e) {
-					// ignore
+	private void parseMulti(Archive archive, File multi) throws Exception {
+		try (Jar jar = new Jar(multi)) {
+			int n = 1000;
+			for (Entry<String, aQute.bnd.osgi.Resource> entry : jar.getResources()
+				.entrySet()) {
+				String path = entry.getKey()
+					.toLowerCase();
+				if (path.endsWith(".jar")) {
+					Archive nextArchive = new Archive(archive.revision, null, "jar",
+						String.format("%s%04d", archive.classifier, n));
+					n++;
+					File dest = repo.toLocalFile(nextArchive);
+					if (!dest.isFile() || dest.lastModified() < multi.lastModified()) {
+						IO.copy(entry.getValue()
+							.openInputStream(), dest);
+						dest.setLastModified(multi.lastModified());
+					}
+					parseSingleOrMultiFile(nextArchive, dest);
 				}
 			}
 		}
+	}
+
+	boolean refresh() throws Exception {
+		if (indexFile.lastModified() != lastModified && last + 10000 < System.currentTimeMillis()) {
```

Comment:
Why the 10sec delay? You are limiting refreshes. Seems restrictive if I want to hit the refresh button in Bndtools.